seqr/views/utils/dataset_utils.py

```diff
@@ -374,17 +386,22 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
             if indiv_id and sample_id not in sample_id_to_individual_id_mapping:
                 sample_id_to_individual_id_mapping[sample_id] = indiv_id
 
+            project = row_dict.pop(PROJECT_COL, None)
+            if project:
+                sample_id_project_tuples.add((sample_id, project))
             samples_by_id[sample_id][gene_id] = row_dict
 
     message = f'Parsed {len(samples_by_id)} RNA-seq samples'
     info = [message]
     logger.info(message, user)
 
     data_source = file_path.split('/')[-1].split('_-_')[-1]
+    projects = Project.objects.filter(name__in={project for _, project in sample_id_project_tuples})
     samples, remaining_sample_ids = _match_and_update_rna_samples(
-        projects=get_internal_projects(),
+        projects=projects if len(projects) > 0 else get_internal_projects(),
         user=user,
         sample_ids=samples_by_id.keys(),
+        sample_id_project_names={f'{sample_id}--{project}' for sample_id, project in sample_id_project_tuples},
```

Comment:
As discussed in the ticket and in the PR I linked there, doing filtration on a concatenated string is not sufficient. You need to add an explicit step that iterates through and confirms that every sample model has the correct project-sample. I recommend something like the set subtraction I added in https://github.com/broadinstitute/seqr/pull/3256/files#diff-a65acc9f2abbf6cf2a54fab4272e559e7bf7f79483a5f66b70e51219417dc84dR125

Also, for this many models I think the `Concat` you added is going to be way too slow for almost no gain. As I explicitly said on the ticket:
> Also, annotating that many samples with the concat field will almost certainly be slower than simply running a query that includes a few duplicates (the query we run now) and then making the sample lookup explicitly include the project

Reply:
Okay, I'll do a sample lookup after the querying.